app/models/player.rb

```diff
@@ -0,0 +1,7 @@
+class Player < ActiveRecord::Base
+	has_one :stat
+	has_many :weight_stats, through: :stats
+	has_many :height_stats, through: :stats
+	has_many :squat_stats, through: :stats
+	has_many :benchpress_stats, through: :stats
```

Comment:
This is where it gets pretty complicated - you can actually just do `has_many stats`, and leave all the other stats. Since the other stats inherit from `stat`, if `stat` belongs to `player,` then all the stats belongs to `player` as well. Does that make sense?